text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
Oh one other thing I forgot to mention is that I believe the primary reason we cannot return an iterator over slices is that on Windows the representation of arguments is UTF-16, but the representation of `OsStr` is not UTF-16. As a result, we must return `OsString` (no matter what, regardless of memory).

To that end the choice of how we handle `argc` and `argv` is basically just an implementation detail and doesn't affect the API too too much.